riposte-spi/src/main/java/com/nike/riposte/server/http/filter/RequestAndResponseFilter.java

```diff
@@ -216,4 +216,17 @@ default boolean isShortCircuitRequestFilter() {
         throw new UnsupportedOperationException(
             "Not implemented - should only be called if isShortCircuitRequestFilter() is true");
     }
+
+    /**
+     *
```

Comment:
The info in this javadoc about how it's used in HttpChannelInitializer and how it relates to the various Netty handlers should probably be removed - it requires a depth of knowledge about the guts of Riposte that most users don't want to or need to know about. Also no need to discuss the backwards compatibility stuff - that's a meta issue for the changelog, not something relevant to users wanting to know what the method does.

I would focus on how this method affects the behavior from a normal-Riposte-user perspective (where the only knobs they know how to tweak come from the `ServerConfig` for their app), and consequences for choosing before vs. after. Something like:

```
This method determines whether the filter should be executed before or after any security validation 
provided by the {@link RequestSecurityValidator} from the application's 
{@link ServerConfig#requestSecurityValidator()}. If configured to execute before security validation 
then *all* requests will be run through this filter. If configured to execute after security validation 
then only requests that pass the security validation will be run through this filter.

@return true if this filter should execute before any {@link ServerConfig#requestSecurityValidator()} 
configured for this application, false if it should execute after 
{@link ServerConfig#requestSecurityValidator()}.
```